content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md

```diff
@@ -0,0 +1,439 @@
+- [Adding a post](#adding-a-post)
+- [Installing the WpGraphQL plugin](#installing-the-wpgraphql-plugin)
+- [Running a query from the GraphiQL IDE](#running-a-query-from-the-graphiql-ide)
+- [Setting up the Next.js environment](#setting-up-the-nextjs-environment)
+- [Adding WordPress API to Next.js using the GraphQL schema](#adding-wordpress-api-to-nextjs-using-the-graphql-schema)
+- [Connecting to the WordPress CMS](#connecting-to-the-wordpress-cms)
+- [Fetch all WordPress posts](#fetch-all-wordpress-posts)
+- [Fetch a single WordPress post](#fetch-a-single-wordpress-post)
+- [Testing](#testing)
+- [Testing with a single post](#testing-with-a-single-post)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### What is a headless CMS
+First, a WordPress CMS, the backend (where you manage your content) and frontend (where the content is displayed to the website visitors) are integrated as one monolithic system. When WordPress runs as a headless CMS, both the content and the content presentation are decoupled. This makes the content presentation layer flexible. You can now build a website, a mobile application, etc to present the raw WordPress-driven content. The backed and the frontend are not tied to the CMS any more.
+
+In simple terms, you still have WordPress CMS for content creation, and management saved to a database. The only difference is that content presentation is delivered through the Application Programming Interface (API) as raw data format such as JSON. Then the data is presented in the websites, mobile applications, or any other channels and devices.
+
+### Reasons for running WordPress as a headless CMS
+- You can change the content delivery channels when you would like to without re-authoring the content.
+
+- Since the content is delivered as raw data through an API, you can use any technology you wish to present this data. As a developer, this allows you things that matter most, such as creating great digital experiences for your users.
+
+- API driven data promote omnichannel architectures. You create WordPress content and channel it through API. Thus the content shines across all touchpoints.
+
+- It creates more efficient, scalable, and fast applications. Both backend and frontend are separated from the monolith CMS architecture. This means if the WordPress-backed CMS has any issues, the channel delivery pipelines are not affected. Hence the performance of the delivery channel is not compromised.
+
+### Adding a post
+
+Since WordPress and Next.js are used to develop blog-based websites, we will use this scenario by first adding and posting a post in WordPress admin.
+
+First, from your WpAdmin dashboard, navigate to `posts` and add a `category`. On the left pane of the resulting page, enter a name, slug, and a description for the post category, and then click the button `Add new category`. Here is how I entered these fields.
+
```

Comment:
A reader not familiar with WordPress would wonder how does one get to the admin dashboard in the first place. You can include a link or explain how one gets to the WpAdmin dashboard